### ISSUE TITLE
CI/CD failure on validating delete node template

### ISSUE THREAD

**Issue by galiacheng** [wls-eng/arm-oraclelinux-wls#199]
We have CI/CD failures in cluster offer which caused by deleting nodes from cluster via delete-node template.

Happens on 12.2.1.4 image since 2020-11-15.

Root cause: the Coherence servers started with exception.  When deleting the server, received error message from Node Manager Server: [WebLogic Server 'managedServer2' is not currently running].

**Comment by galiacheng** [comment]
Error: 
```text

<Dec 10, 2020 10:11:52,768 AM UTC> <Error> <CoherenceIntegration> <BEA-2194507> <The Coherence cluster service failed to start or failed to join cluster due to (Wrapped) java.lang.reflect.InvocationTargetException using Exception configuring coherence..>
<Dec 10, 2020 10:11:55,720 AM UTC> <Warning> <JMX> <BEA-149513> <JMX Connector Server stopped at service:jmx:iiop://mspStorageVM1:7501/jndi/weblogic.management.mbeanservers.runtime.>
<Dec 10, 2020 10:11:55,781 AM UTC> <Critical> <WebLogicServer> <BEA-000362> <Server failed. Reason:

There are 1 nested errors:

weblogic.coherence.api.internal.CoherenceException: (Wrapped) null
        at weblogic.cacheprovider.coherence.CoherenceClusterManager.processException(CoherenceClusterManager.java:837)
        at weblogic.cacheprovider.coherence.CoherenceClusterManager.configureClusterService(CoherenceClusterManager.java:291)
        at weblogic.cacheprovider.CacheProviderServerService.bootCoherenceFromWLSCluster(CacheProviderServerService.java:229)
        at weblogic.cacheprovider.CacheProviderServerService.initCoherence(CacheProviderServerService.java:103)
        at weblogic.cacheprovider.CacheProviderServerService.initialize(CacheProviderServerService.java:80)
        at weblogic.cacheprovider.CacheProviderServerService.start(CacheProviderServerService.java:74)
        at weblogic.server.AbstractServerService.postConstruct(AbstractServerService.java:76)
        at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke(ReflectionHelper.java:1287)
        at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:333)
        at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:375)
        at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:487)
        at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:305)
        at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:85)
        at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2126)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:116)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:90)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1237)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1168)
        at weblogic.work.SelfTuningWorkManagerImpl$WorkAdapterImpl.run(SelfTuningWorkManagerImpl.java:681)
        at weblogic.invocation.ComponentInvocationContextManager._runAs(ComponentInvocationContextManager.java:352)
        at weblogic.invocation.ComponentInvocationContextManager.runAs(ComponentInvocationContextManager.java:337)
        at weblogic.work.LivePartitionUtility.doRunWorkUnderContext(LivePartitionUtility.java:57)
        at weblogic.work.PartitionUtility.runWorkUnderContext(PartitionUtility.java:41)
        at weblogic.work.SelfTuningWorkManagerImpl.runWorkUnderContext(SelfTuningWorkManagerImpl.java:655)
        at weblogic.work.ExecuteThread.execute(ExecuteThread.java:420)
        at weblogic.work.ExecuteThread.run(ExecuteThread.java:360)
Caused by: (Wrapped) java.lang.reflect.InvocationTargetException
        at com.tangosol.util.Base.ensureRuntimeException(Base.java:324)
        at com.tangosol.util.Base.ensureRuntimeException(Base.java:305)
        at com.tangosol.net.CacheFactory.setServiceConfig(CacheFactory.java:1028)
at com.tangosol.net.CacheFactory.setManagementConfig(CacheFactory.java:969)
        at weblogic.coherence.service.internal.WLSCoherenceConfigurator.injectConfiguration(WLSCoherenceConfigurator.java:421)
        at weblogic.coherence.service.internal.WLSCoherenceConfigurator.configure(WLSCoherenceConfigurator.java:354)
        at weblogic.coherence.service.internal.WLSCoherenceConfigurator.configureCoherence(WLSCoherenceConfigurator.java:147)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at weblogic.cacheprovider.coherence.CoherenceClusterManager.configureClusterService(CoherenceClusterManager.java:287)
        ... 27 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.tangosol.net.CacheFactory.setServiceConfig(CacheFactory.java:1024)
        ... 36 more
Caused by: java.lang.IllegalArgumentException: MBean type=Cluster,cluster=myCoherence does not exist
        at com.tangosol.coherence.component.net.management.gateway.Remote$LocalMBeanServerProxy.getMBeanInfo(Remote.CDB:9)
        at com.tangosol.internal.metrics.MetricSupport.registerInternal(MetricSupport.java:196)
        at com.tangosol.internal.metrics.MetricSupport.register(MetricSupport.java:139)
        at com.tangosol.coherence.component.net.management.Gateway.registerMetrics(Gateway.CDB:1)
        at com.tangosol.coherence.component.net.management.gateway.Remote.registerLocalModel(Remote.CDB:10)
        at com.tangosol.coherence.component.net.management.Gateway.register(Gateway.CDB:8)
        at com.tangosol.coherence.component.net.management.Gateway.makePrimary(Gateway.CDB:6)
        at com.tangosol.coherence.component.net.management.Gateway.createGateway(Gateway.CDB:91)
        at com.tangosol.coherence.component.application.console.Coherence.setServiceConfig(Coherence.CDB:32)
        ... 41 more
```

**Comment by galiacheng** [comment]
Error: 
```
Dec 11 04:16:24 managedServerStorageVM1 startNodeManager.sh[13171]: <Dec 11, 2020 4:16:24 AM UTC> <WARNING> <Server start command for WebLogic server 'managedServerStorage1' failed due to: [Server failed to start up but Node Manager was not aware of the reason]. Please check Node Manager log and/or server 'managedServerStorage1' log for detailed information.>
Dec 11 04:28:18 managedServerStorageVM1 systemd[1]: [/etc/systemd/system/wls_nodemanager.service:10] Not an absolute path, ignoring: "/u01/domains/wlsd"
Dec 11 04:28:18 managedServerStorageVM1 systemd[1]: [/etc/systemd/system/wls_nodemanager.service:10] Not an absolute path, ignoring: "/u01/domains/wlsd"
Dec 11 04:30:03 managedServerStorageVM1 systemd[1]: [/etc/systemd/system/wls_nodemanager.service:10] Not an absolute path, ignoring: "/u01/domains/wlsd"
Dec 11 04:30:03 managedServerStorageVM1 systemd[1]: [/etc/systemd/system/wls_nodemanager.service:10] Not an absolute path, ignoring: "/u01/domains/wlsd"
Dec 11 04:36:16 managedServerStorageVM1 systemd[1]: [/etc/systemd/system/wls_nodemanager.service:10] Not an absolute path, ignoring: "/u01/domains/wlsd"
```

**Comment by galiacheng** [comment]
Node manager log:
```
<Dec 11, 2020 4:15:18 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Starting WebLogic server with command line: /u01/domains/wlsd/bin/startWebLogic.sh >
<Dec 11, 2020 4:15:18 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Working directory is '/u01/domains/wlsd'>
<Dec 11, 2020 4:15:18 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server output log file is '/u01/domains/wlsd/servers/managedServerStorage1/logs/manag
edServerStorage1.out'>
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <The server 'managedServerStorage1' with process id 13349 is no longer alive; waiting
for the process to die.>
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server failed during startup. It may be retried according to the auto restart configu
ration.>
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server failed so attempting to restart (restart count = 1)>
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Starting WebLogic server with command line: /u01/domains/wlsd/bin/startWebLogic.sh >
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Working directory is '/u01/domains/wlsd'>
<Dec 11, 2020 4:15:44 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server output log file is '/u01/domains/wlsd/servers/managedServerStorage1/logs/manag
edServerStorage1.out'>
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <The server 'managedServerStorage1' with process id 13491 is no longer alive; waiting
for the process to die.>
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server failed during startup. It may be retried according to the auto restart configu
ration.>
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server failed so attempting to restart (restart count = 2)>
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Starting WebLogic server with command line: /u01/domains/wlsd/bin/startWebLogic.sh >
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Working directory is '/u01/domains/wlsd'>
<Dec 11, 2020 4:16:04 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server output log file is '/u01/domains/wlsd/servers/managedServerStorage1/logs/manag
edServerStorage1.out'>
<Dec 11, 2020 4:16:24 AM UTC> <INFO> <wlsd> <managedServerStorage1> <The server 'managedServerStorage1' with process id 13643 is no longer alive; waiting
for the process to die.>
<Dec 11, 2020 4:16:24 AM UTC> <INFO> <wlsd> <managedServerStorage1> <Server failed during startup. It may be retried according to the auto restart configuration.>
```

**Comment by galiacheng** [comment]
From this page [Oracle Weblogic 12.2.1.4 - WLS PATCH SET UPDATE 12.2.1.4.191220 fix Coherence Exception (Wrapped) null](https://blog.dba.bg/index.php/kkovachki/oracle/oracle-weblogic-12-2-1-4-wls-patch-set-update-12-2-1-4-191220-fix-coherence-exception-wrapped-null), we should apply p30729380_122140_Generic.zip.

**Comment by galiacheng** [comment]
[After Applying Patch 30689820(WLS 12.2.1.4.191220) Managed Server Fails To Start Due To Coherence Errors](https://support.oracle.com/epmos/faces/DocumentDisplay?_afrLoop=423437838176196&parent=EXTERNAL_SEARCH&sourceId=PROBLEM&id=2634266.1&_afrWindowMode=0&_adf.ctrl-state=196qh5zzf1_409)

**Comment by galiacheng** [comment]
After applied the above patches, the Coherence servers start without exception.

**Comment by galiacheng** [comment]
We have two options to fix this issue:
1. Apply the patches to base image `WebLogic Server 12.2.1.4.0 Base Image and JDK8u251 on OL7.6`
2. Apply the patches via VM extension dynamically

   - Have to ask customer a oracle support account to download the patches, or provide create a default account.

   - Apply the patches via OPatch

**Comment by galiacheng** [comment]
Close this issue, as we will fix it in #202
